src/test/java/com/stripe/functional/FeeRefundTest.java

```diff
@@ -0,0 +1,85 @@
+package com.stripe.functional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.stripe.BaseStripeTest;
+import com.stripe.exception.StripeException;
+import com.stripe.model.ApplicationFee;
+import com.stripe.model.FeeRefund;
+import com.stripe.model.FeeRefundCollection;
+import com.stripe.net.APIResource;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+public class FeeRefundTest extends BaseStripeTest {
+  public static final String FEE_ID = "fee_123";
+  public static final String REFUND_ID = "fr_123";
+
+  @Test
+  public void testCreate() throws StripeException {
+    final ApplicationFee fee = ApplicationFee.retrieve(FEE_ID);
+
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("amount", 100);
+
+    final FeeRefund refund = fee.getRefunds().create(params);
+
+    assertNotNull(refund);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/application_fees/%s/refunds", fee.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testRetrieve() throws StripeException {
+    final ApplicationFee fee = ApplicationFee.retrieve(FEE_ID);
+
+    final FeeRefund refund = fee.getRefunds().retrieve(REFUND_ID);
+
+    assertNotNull(refund);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/application_fees/%s/refunds/%s", fee.getId(), REFUND_ID)
+    );
+  }
+
+  @Test
+  public void testUpdate() throws StripeException {
+    final ApplicationFee fee = ApplicationFee.retrieve(FEE_ID);
+    final FeeRefund refund = fee.getRefunds().retrieve(REFUND_ID);
+
+    Map<String, Object> metadata = new HashMap<String, Object>();
+    metadata.put("key", "value");
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("metadata", metadata);
+
+    final FeeRefund updatedRefund = refund.update(params);
+
+    assertNotNull(updatedRefund);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/application_fees/%s/refunds/%s", fee.getId(), refund.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testList() throws StripeException {
+    final ApplicationFee fee = ApplicationFee.retrieve(FEE_ID);
+
+    final FeeRefundCollection reversals = fee.getRefunds().list(null);
```

Comment:
rename the variable to `resources` instead (bad copy paste I think)

Reply:
Ah yeah, this one should be `refunds`. (I understand that using `resource` / `resources` everywhere makes the initial effort of porting/writing tests easier, but I prefer explicit names.)

Reply:
> I understand that using resource / resources everywhere makes the initial effort of porting/writing tests easier, but I prefer explicit names.

I disagree, that avoids the copy/paste fails we have in most of our libraries. But would be great to be consistent since we already went through so much trouble to make everything consistent.